includes/SpecialRemovePII.php

```diff
@@ -0,0 +1,223 @@
+
+	/**
+	 * @return array|string
+	 */
+	protected function getFormFields() {
+		if (
+			$this->config->get( 'RemovePIICentralWiki' ) &&
+			$this->config->get( 'DBname' ) !== $this->config->get( 'RemovePIICentralWiki' )
+		) {
+			return $this->msg( 'removepii-wiki-disabled' )->escaped();
+		}
+
+		$formDescriptor = [];
+		
+		$formDescriptor['warning'] = [
+			'type' => 'info',
+			'help' => Html::warningBox( $this->msg( 'removepii-warning-irreversible' )->parse() ),
+			'raw-help' => true,
+			'hide-if' => [ '!==', 'wpaction', 'removepii' ]
+		];
+
+		$formDescriptor['oldname'] = [
+			'type' => 'text',
+			'required' => true,
+			'label-message' => 'removepii-oldname-label'
+		];
+
+		$formDescriptor['newname'] = [
+			'type' => 'text',
+			'required' => true,
+			'label-message' => 'removepii-newname-label'
+		];
+		
+		$formDescriptor['action'] = [
+			'type' => 'select',
+			'options' => [
+				'Rename user' => 'renameuser',
+				'RemovePII' => 'removepii'
+			],
+			'required' => true,
+			'default' => 'renameuser',
+			'label-message' => 'removepii-action-label',
+			'help-message' => 'removepii-action-help'
+		];
+
+		return $formDescriptor;
+	}
+
+	/**
+	 * @param array $formData
+	 * @return Status
+	 */
+	public function validate( array $formData ) {
+		if ( !ExtensionRegistry::getInstance()->isLoaded( 'CentralAuth' ) ) {
+			return Status::newFatal( 'removepii-centralauth-notinstalled' );
+		}
+
+		if ( !ExtensionRegistry::getInstance()->isLoaded( 'Renameuser' ) ) {
+			return Status::newFatal( 'centralauth-rename-notinstalled' );
+		}
+
+		$oldUser = User::newFromName( $formData['oldname'] );
+		if ( !$oldUser ) {
+			return Status::newFatal( 'centralauth-rename-doesnotexist' );
+		}
+
+		if ( $oldUser->getName() === $this->getUser()->getName() ) {
+			return Status::newFatal( 'centralauth-rename-cannotself' );
+		}
+
+		$newUser = User::newFromName( $formData['newname'] );
+		if ( !$newUser ) {
+			return Status::newFatal( 'centralauth-rename-badusername' );
+		}
+
+		$validator = new GlobalRenameUserValidator();
+		$status = $validator->validate( $oldUser, $newUser );
+
+		return $status;
+	}
+
+	/**
+	 * @param array $formData
+	 * @return bool
+	 */
+	public function onSubmit( array $formData ) {
+		$out = $this->getOutput();
+
+		if ( $formData['action'] === 'renameuser' ) {
+			$valid = $this->validate( $formData );
+			if ( !$valid->isOK() ) {
+				return $valid;
+			}
+
+			$oldUser = User::newFromName( $formData['oldname'] );
+			$newUser = User::newFromName( $formData['newname'], 'creatable' );
+
+			$session = $this->getContext()->exportSession();
+			$globalRenameUser = new GlobalRenameUser(
+				$this->getUser(),
+				$oldUser,
+				CentralAuthUser::getInstance( $oldUser ),
+				$newUser,
+				CentralAuthUser::getInstance( $newUser ),
+				new GlobalRenameUserStatus( $newUser->getName() ),
+				'JobQueueGroup::singleton',
+				new GlobalRenameUserDatabaseUpdates(),
+				new GlobalRenameUserLogger( $this->getUser() ),
```

Comment:
Just adding a couple personal notes. Can we create a new class in this extension, which extends this `GlobalRenameUserLogger` class and override the `log()` function to `return false;` to never log renames done here?

Reply:
This has been done.